### ISSUE TITLE
Vernacular names: only allow data entry for those flagged as "accepted name"

### ISSUE THREAD

**Issue by kitenetter** [BiologicalRecordsCentre/iRecord#665]
This arose as a side-issue from #163 and relates to #636 

The multiplicity of vernacular names in the UKSI can cause problems, e.g. a record imported as "Emperor" gets linked to the moth rather than the dragonfly, but "Emperor" is flagged in UKSI as a "junior synonym" (http://nbn-sd-dev.nhm.ac.uk/taxon.php?linkKey=NBNSYS0000166745) while "Emperor Moth" is flagged as "accepted name" (http://nbn-sd-dev.nhm.ac.uk/taxon.php?linkKey=NHMSYS0000502354)

We should review the list of vernacular names in the new version of the UKSI and see if we can restrict the use of names that are not flagged as "accepted name" (I'm not certain how consistently the "accepted name" flag has been applied).

Question: if we mark vernacular names in the iRecord dictionary as "not for data entry" does that restrict their use during csv upload (which is what is needed) as well as during data entry via forms?

**Comment by DavidHepper** [comment]
We have another case today where a user has imported multiple "Emperor" instead of "Emperor Dragonfly" and these have all become moth records, invisible to the dragonfly verifiers. Please reject "Emperor".

**Comment by kitenetter** [comment]
@DavidHepper do you need us to delete the imported records, or has the user done so already? If you do need us to do so can you let me know the username and any indiciation of how many records, the record ID range, or the date of import.

**Comment by sacrevert** [comment]
@kitenetter can this be closed? It seems to be partly a UKSI maintenance issue (which I would have thought is for schemes to manage with Chris R), and separately an issue of how specific surveys or forms might be set up.

**Comment by DavidHepper** [comment]
Be aware that in UKSI many Odonata have English, Welsh and Scottish common names. Only one of each language should be flagged 'accepted name'.

**Comment by sacrevert** [comment]
> Be aware that in UKSI many Odonata have English, Welsh and Scottish common names. Only one of each language should be flagged 'accepted name'.

As I mentioned, this is typically something that we leave to schemes to organise with Chris R directly unless we have resources and have been specifically asked to sort out some UKSI related issue for a scheme. For instance, it seems that the Emporer dragonfly has more than one Welsh name with accepted status (https://uksi-sandbox.nhm.ac.uk/taxon.php?linkKey=NBNSYS0000166746); if you want this changed you should liaise with Chris. He has spreadsheets in place that schemes can work with to provide him with updates to the UKSI.

**Comment by DavidHepper** [comment]
Yes, and currently Norfolk Hawker is being shown as Norfolk Aeshna in iRecord, a name "so old it has dust on it", because neither was marked 'accepted name' and it defaulted to the oldest! This should be fixed in the next minor update from Chris.

**Comment by sacrevert** [comment]
To recentre this thread on the original issue, it seems that the main question is still "Question: if we mark vernacular names in the iRecord dictionary as "not for data entry" does that restrict their use during csv upload (which is what is needed) as well as during data entry via forms?" as per the original post. (See opening post for more detail)

**Comment by johnvanbreda** [comment]
No, we don't restrict "not for data entry" for imports, I recall a long while back that this was deliberate due to the need to import data from things like literature, with old names. But I agree that it could be an option that needs to be enabled.

@MarkAlliston The NHM sandbox data for Norfolk Hawker (https://uksi-sandbox.nhm.ac.uk/taxon.php?linkKey=NHMSYS0021775230) has a flag "accepted name" next to Norfolk Hawker. However the data we have imported from UKSI (in uksi.all_names) has identical flags, we don't seem to be receiving anything that indicates which is an accepted common name. I don't think there was such a thing in UKSI a few years back but maybe it's been added - could you check the data held in the Access database please to see if you can spot where this flag comes from? At the moment, the script "28 - prepare common names.sql" is unable to detect a preference between Norfolk Hawker and Norfolk Aeshna so just picks the first.

**Comment by DavidHepper** [comment]
The addition of the 'accepted name' flag to 'Norfolk Hawker' in Sandbox was only made yesterday (02/06/2026).

**Comment by MarkAlliston** [comment]
The best bet would be if I email Chris, I cannot see a field that says Accepted Name in the Access Database Taxon table. ( I have looked in other tables as well with no luck)

**Comment by DavidHepper** [comment]
@MarkAlliston - That's as expected. I had to ask for a correction. See https://forums.nbn.org.uk/viewtopic.php?id=8068

**Comment by MarkAlliston** [comment]
@johnvanbreda its in the NAMESERVER table and called TAXON_VERSION_STATUS = R

it gets pulled into the all_names txt file

**Comment by johnvanbreda** [comment]
@MarkAlliston I'm away from my desk at the moment so won't connect to the live servers to check, but doesn't Norfolk Aeshna also have TAXON_VERSION_STATUS='R'? That's what I thought I observed when I looked at it earlier.

**Comment by MarkAlliston** [comment]
No it looks like Norfolk Aeshena is S 

We can have two names as accepted_names but it it looks like they have to be different name types i.e.  scientific and vernacular

**Comment by DavidHepper** [comment]
The 'Norfolk Aeshna' problem is still a potential cause of confusion. Can this be patched or are you planning a minor update of UKSI in iRecord? If so, when?

**Comment by DavidHepper** [comment]
Example: https://irecord.org.uk/record-details/verification?occurrence_id=55427954
